adafruit_httpserver/server.py

```diff
@@ -2,97 +2,220 @@
 #
 # SPDX-License-Identifier: MIT
 """
-`adafruit_httpserver.server.HTTPServer`
+`adafruit_httpserver.server`
 ====================================================
 * Author(s): Dan Halbert, Michał Pokusa
 """
 
 try:
-    from typing import Callable, Protocol, Union
+    from typing import Callable, Protocol, Union, List, Set, Tuple
     from socket import socket
     from socketpool import SocketPool
 except ImportError:
     pass
 
 from errno import EAGAIN, ECONNRESET, ETIMEDOUT
-
-from .exceptions import FileNotExistsError, InvalidPathError
-from .methods import HTTPMethod
-from .request import HTTPRequest
-from .response import HTTPResponse
-from .route import _HTTPRoutes, _HTTPRoute
-from .status import CommonHTTPStatus
-
-
-class HTTPServer:
+from traceback import print_exception
+
+from .authentication import Basic, Bearer, require_authentication
+from .exceptions import (
+    ServerStoppedError,
+    AuthenticationError,
+    FileNotExistsError,
+    InvalidPathError,
+    ServingFilesDisabledError,
+)
+from .methods import GET, HEAD
+from .request import Request
+from .response import Response, FileResponse
+from .route import _Routes, _Route
+from .status import BAD_REQUEST_400, UNAUTHORIZED_401, FORBIDDEN_403, NOT_FOUND_404
+
+
+class Server:
     """A basic socket-based HTTP server."""
 
-    def __init__(self, socket_source: Protocol, root_path: str) -> None:
+    host: str = None
+    port: int = None
+
+    def __init__(
+        self, socket_source: Protocol, root_path: str = None, *, debug: bool = False
+    ) -> None:
         """Create a server, and get it ready to run.
 
         :param socket: An object that is a source of sockets. This could be a `socketpool`
           in CircuitPython or the `socket` module in CPython.
         :param str root_path: Root directory to serve files from
+        :param bool debug: Enables debug messages useful during development
         """
+        self._auths = []
         self._buffer = bytearray(1024)
         self._timeout = 1
-        self.routes = _HTTPRoutes()
+        self._routes = _Routes()
         self._socket_source = socket_source
         self._sock = None
         self.root_path = root_path
```

Comment:
While the current behavior around root_path is much improved from prior versions, I do think it's worth adding one additional warning to the user in the form of a print statement if they do set `root_path` to `/` 

Maybe something like this:

```
if self.root_path == "/":
    print("WARNING: Setting root_path to '/' will expose all files on your device through the webserver, including potentially sensitive files like settings.toml or secrets.py. Consider making a sub-directory on your device and using that for your root_path instead.")
```

Reply:
I added the warning when `root_path` is set to `"/"` or `""`, which results in the same behavior as `"/"`.

Reply:
Nice! I tested the new warning successfully. Thanks for adding that. (And again for all of your work on this library!)